quixstreams/state/base/transaction.py

```diff
@@ -348,11 +398,32 @@ def set(self, key: K, value: V, prefix: bytes, cf_name: str = "default"):
         """
 
         try:
-            key_serialized = self._serialize_key(key, prefix=prefix)
             value_serialized = self._serialize_value(value)
+        except Exception:
+            self._status = PartitionTransactionStatus.FAILED
+            raise
+
+        self.set_bytes(key, value_serialized, prefix, cf_name=cf_name)
+
+    @validate_transaction_status(PartitionTransactionStatus.STARTED)
+    def set_bytes(
+        self, key: K, value: bytes, prefix: bytes, cf_name: str = "default"
+    ) -> None:
+        """
+        Set bytes value for the key.
+        :param key: key
+        :param prefix: a key prefix
+        :param value: value
+        :param cf_name: column family name
+        """
+        try:
+            if not isinstance(value, bytes):
```

Comment:
is this check necessary? type hint is strict here

Reply:
Daniil wanted me to add it. Without it, passing an invalid value will only raise when the checkpoint is committed. That's much harder to debug